### ISSUE TITLE
iOS game crash for random maps while starting multiplayer

### ISSUE THREAD

**Issue by Nordsoft91** [vcmi/vcmi#984]
### Steps
1. Download and install exactly the same build on iOS device and MacOS device. I used this one https://github.com/vcmi/vcmi/actions/runs/3118396255
2. Start game from both devices being in the same local network without mods (except extras and essentials)
3. Start multiplayer MacOS as host, iOS joins.
4. You will see devices connected in the lobby and everything works there
5. Launch any random map
6. Observe game crash on iOS

### Observations
- Issue appears for random maps. For scenarios everything works well
- I haven't network mode between any other device types
- If you launch 2 application instances locally and start multiplayer using 127.0.0.1, there is no problem in starting random maps
- If iOS will be game host, there will be immediate message that iOS player lost instead of crash

**Platform-dependent floating point computations in RMG?**

**Comment by kambala-decapitator** [comment]
using latest nightly build, existing scenario (not even rmg). Additionally enabled Campaign Heroes mod. Players: macOS 12.6 with Intel CPU, arm64 iOS 15.7, x86 Windows 8.1. Only iOS is on wireless connection, if it matters.

test 1: macOS host, iOS client, Windows client - windows crashed on starting game. [windows.txt](https://github.com/vcmi/vcmi/files/9640467/windows.txt)

test 2: windows host, iOS tries to connect - iOS crash

test 3: windows host, macOS connects, iOS connects - iOS hangs (not crashes) on starting game, seems it's stuck here:

```
* thread #13, name = 'CServerHandler::threadHandleConnection'
  * frame #0: 0x00000001bb673e94 libsystem_kernel.dylib`__recvfrom + 8
    frame #1: 0x0000000105133d6c libvcmi_lib_client.dylib`boost::asio::detail::socket_ops::sync_recv1(int, unsigned char, void*, unsigned long, int, boost::system::error_code&) + 212
    frame #2: 0x000000010512814c libvcmi_lib_client.dylib`LIB_CLIENT::CConnection::read(void*, unsigned int) + 128
    frame #3: 0x000000010513440c libvcmi_lib_client.dylib`void LIB_CLIENT::BinaryDeserializer::load<LIB_CLIENT::CPack*, 0>(LIB_CLIENT::CPack*&) + 56
    frame #4: 0x0000000105128938 libvcmi_lib_client.dylib`LIB_CLIENT::CConnection::retrievePack() + 60
    frame #5: 0x0000000100f508f4 vcmiclient`CServerHandler::threadHandleConnection() + 284
    frame #6: 0x00000001048c5d10 libboost_thread.dylib`boost::(anonymous namespace)::thread_proxy(void*) + 232
    frame #7: 0x00000001dbcca348 libsystem_pthread.dylib`_pthread_start + 116
```

test 4: iOS host, macOS joins, windows managed to join only in a couple of minutes (probably because iOS server isn't that fast) - windows hangs on starting game (resembles test 3). After exiting game with server, windows correctly goes back to main menu.

test 5: windows host, iOS connects, macOS connects - macOS hangs on starting game, AI controls that player. During AI turn windows crashes.

test 6 with RMG (XL map, everything random): macOS host, iOS client, Windows client - when host's (red) turn started, both clients were still sitting on the loading screen. Then it was iOS (blue) turn, some seconds after the hourglass started on macOS, iOS finally joined and worked normally, at the same time windows client crashed and "your heroes abandon you" appeared. [windows Log + crash dump](https://github.com/vcmi/vcmi/files/9640484/Archive.zip)

**Comment by kambala-decapitator** [comment]
I guess this will be fixed by #1000?
